crates/biome_html_analyze/src/lint/nursery/use_vue_v_for_key.rs

```diff
@@ -0,0 +1,118 @@
+        language: "html",
+        recommended: true,
+        domains: &[RuleDomain::Vue],
+        sources: &[RuleSource::EslintVueJs("require-v-for-key").same()],
+    }
+}
+
+impl Rule for UseVueVForKey {
+    type Query = Ast<HtmlAttributeList>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = UseVueVForKeyOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let attrs = ctx.query();
+        let mut has_v_for = None;
+        let mut has_key = false;
+        for attr in attrs.iter() {
+            if let Some(dir_any) = attr.as_any_vue_directive() {
+                match dir_any {
+                    AnyVueDirective::VueDirective(dir) => {
+                        if dir.name_token().is_ok_and(|t| t.text_trimmed() == "v-for") {
+                            has_v_for = Some(dir.range());
+                        }
+                        if dir.name_token().is_ok_and(|t| t.text_trimmed() == "v-bind")
+                            && let Some(arg) = dir.arg()
+                            && is_v_bind_key_argument(&arg)
+                        {
+                            has_key = true;
+                        }
+                    }
+                    AnyVueDirective::VueVBindShorthandDirective(sh) => {
+                        if let Ok(arg) = sh.arg()
+                            && is_v_bind_key_argument(&arg)
+                        {
+                            has_key = true;
+                        }
+                    }
+                    _ => {}
+                }
+                if has_v_for.is_some() && has_key {
+                    // early exit if both found
+                    return None;
+                }
+            }
+        }
+        if let (Some(v_for_range), false) = (has_v_for, has_key) {
+            return Some(v_for_range);
+        }
+        None
+    }
+
+    fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                state,
+                markup! {
+                    "This element is using "<Emphasis>"v-for"</Emphasis>", but the "<Emphasis>"key"</Emphasis>" is missing."
+                },
+            )
+            .note(markup! {
+                "Using a unique key with "<Emphasis>"v-for"</Emphasis>" helps Vue optimize rendering and track elements efficiently. Failing to provide a key can result in unexpected behavior during updates. For example, if you want to do transition animations, using a key is necessary so Vue knows which element to apply the animation to."
+            })
+            .note(markup! {
+                "Provide the key using "<Emphasis>":key=\"value\""</Emphasis>", and have the value be a unique value from the items you are iterating over. For example: `<li v-for=\"item in items\" :key=\"item.id\">{{ item }}</li>`"
+            }),
```

Comment:
Any chance we can reduce the description, and add a link to the Vue docs instead? I would remove the "Fox example.." phrase.  That link could be added to the docs too